test/e2e/node/stress_probe.go

```diff
@@ -0,0 +1,194 @@
+	*/
+	ginkgo.It("should *not* be restarted with a GRPC liveness probe stress test with MANY containers", func(ctx context.Context) {
+		livenessProbe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				GRPC: &v1.GRPCAction{
+					Port:    containerPort,
+					Service: nil,
+				},
+			},
+			InitialDelaySeconds: probeTestInitialDelaySeconds,
+			TimeoutSeconds:      5, // default 1s can be pretty aggressive in CI environments with low resources
+			FailureThreshold:    1,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: int32(containerPort)}}
+		probeArgs := []string{"grpc-health-checking", "--port", "", "--http-port", ""}
+		pod := livenessPodSpec("grpc", f.Namespace.Name, livenessProbe, ports, probeArgs...)
+		RunLivenessTest(ctx, f, pod, defaultObservationTimeout)
+	})
+})
+
+func livenessPodSpec(probeType, namespace string, livenessProbe v1.Probe, ports []v1.ContainerPort, args ...string) *v1.Pod {
+	pod := e2epod.NewAgnhostPod(namespace, "liveness-stress-"+string(uuid.NewUUID()), nil, nil, ports, args...)
+	pod.ObjectMeta.Labels = map[string]string{"test": "liveness-stress"}
+	pod.Spec.Containers[0].LivenessProbe = &livenessProbe
+	pod.Spec.Containers[0].Name = "agnhost-container-0"
+	pod.Spec.Containers = append(pod.Spec.Containers, make([]v1.Container, numContainers-1)...)
+	for idx := range make([]int, numContainers) {
+		newContainerPort := idx + containerPort + 1
+		pod.Spec.Containers[idx] = *pod.Spec.Containers[0].DeepCopy()
+		pod.Spec.Containers[idx].Name = "agnhost-container-" + strconv.Itoa(idx+1)
+		pod.Spec.Containers[idx].Ports = []v1.ContainerPort{{ContainerPort: int32(newContainerPort)}}
+		switch probeType {
+		case "tcp":
+			pod.Spec.Containers[idx].LivenessProbe.TCPSocket.Port = intstr.FromInt(newContainerPort)
+			pod.Spec.Containers[idx].Args[3] = strconv.Itoa(newContainerPort)
+		case "http":
+			pod.Spec.Containers[idx].LivenessProbe.HTTPGet.Port = intstr.FromInt(newContainerPort)
+			pod.Spec.Containers[idx].Args[3] = strconv.Itoa(newContainerPort)
+		case "grpc":
+			pod.Spec.Containers[idx].LivenessProbe.GRPC.Port = int32(newContainerPort)
+			pod.Spec.Containers[idx].Args[2] = strconv.Itoa(newContainerPort)
+			pod.Spec.Containers[idx].Args[4] = strconv.Itoa(newContainerPort + containerPort)
+		}
+	}
+	return pod
+}
+
+func httpGetHandler(path string, port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		HTTPGet: &v1.HTTPGetAction{
+			Path: path,
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func tcpSocketHandler(port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		TCPSocket: &v1.TCPSocketAction{
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+// RunLivenessTest verifies the number of restarts.
+func RunLivenessTest(ctx context.Context, f *framework.Framework, pod *v1.Pod, timeout time.Duration) {
+	deadline := time.Now().Add(timeout)
```

Comment:
you should not take into account the time to create pod since is not deterministic